### DISCUSSION TITLE
Command block not working

### DISCUSSION THREAD

**Discussion by AndroXStudiO** [Sketchware-Pro/Sketchware-Pro#560] [Q&A] [answered]
i am trying to change getContext().getApplicationContext() to getActivity()

but Only for this codes command block not working. but if i want to change any other codes then its working. i dnt know why i am facing this issue. any other better option instead Command block?

here my example :

![IMG_20220830_103443](https://user-images.githubusercontent.com/88816748/187350082-c7ada652-abcc-4b7a-9e81-fe751ed0bad7.jpg)
![IMG_20220830_103042](https://user-images.githubusercontent.com/88816748/187350100-f1ea5008-c5b9-4ac8-9169-de421f6dfd19.jpg)

**Comment by j-px** [comment]
Can you show your whole Command Block? By the way, you can save that single Block, and create a clean screenshot of just the Block when in Collections > Block.

**Comment by AndroXStudiO** [comment]
how to show full block?
LogicEditor page not zoom-able.

anyway see this pic 
![Screenshot_2022-08-30-19-22-11-17_423e58f25c30a02453fa5d8c82fc37f2](https://user-images.githubusercontent.com/88816748/187448587-45e96b2a-26cf-4060-af3b-8bf79fd1a329.jpg)

**Comment by AndroXStudiO** [comment]
![Screenshot_2022-08-30-19-25-40-93_423e58f25c30a02453fa5d8c82fc37f2](https://user-images.githubusercontent.com/88816748/187449033-30714f89-135e-4ecc-a655-d54400d09d81.jpg)


already i have tried > find-replace
>replace
>find replace all


i means all options.

**Comment by j-px** [comment, marked as answer]
Found the fix! Since `find-replace` and the other 2 similar commands use regular expressions, you must escape special characters like `(` with `\`. This Block works:

![bro](https://user-images.githubusercontent.com/40118124/187452158-be8e6a55-91ff-4414-83c9-b8d03e5b1161.png)

By the way, this is intended behavior. Even v6.3.0 fix1 does this.

**Comment by j-px** [comment]
@AndroXStudiO did it work? If yes, please tell, so I can "close" this question Issue.

**Comment by AndroXStudiO** [comment]
let me check

**Comment by AndroXStudiO** [comment]
YES. now WORKING with >fine-replace-all
and using escape special characters like \(\)\

**Comment by ilyassesalama** [comment]
Just gave it a try and it doesn't seem to be working, weird. Also why Sketchware Pro doesn't escape those chars automatically like vanilla Sketchware does for some expressions?

**Reply by j-px** [reply]
Did you make sure to use find-replace-* and escape regex syntax characters?

Sketchware [Pro] escapes characters for them to be valid in Java code / XML, not to be valid in regex cases like these.

**Reply by ilyassesalama** [reply]
Here's what I tried (tried multiple ways also). For the record, command blocks used to work just fine in 6.3.0 fix1 for instance. Even without escaping anything.

![Screenshot_20220930-210002_Sketchware Pro~2](https://user-images.githubusercontent.com/46769766/193347481-4d8c305e-d02e-47e2-a7b2-a33db10f0b07.png)

**Reply by j-px** [reply]
Just installed v6.3.0 fix1 to test that out, and the same Command Block that works in v6.4.0 worked there too. Note that I used "find-replace-all".

**Reply by j-px** [reply]
Just "find-replace" seems to not be using regex. Adjusted the Block to work in v6.3.0 fix1 with that mode, installed v6.4.0, and got the same changes there.

**Reply by ilyassesalama** [reply]
Guess what, just installed "v6.4.0 test build 04" and even the following command block works just fine.
![Screenshot_20220930-212246_Sketchware Pro~2](https://user-images.githubusercontent.com/46769766/193350761-46913e11-bbb9-4cc2-a20b-2578a60437c1.png)

By the way, it seems like custom events are not working in the latest actions' version as well.

If you want to give it a try, here's a custom event to add custom variables which is working in the build mentioned above, but not in the newest GitHub actions' version:

```
/*-JX4UA2y_f1OckjjvxWI.bQwRei-sLEsBmds7ArsRfi0xSFEP3Php97kjdMCs5ed
>["public class"]
>1
>0
>0
>add
%s
BpWI8U4flOpx8Ke66QTlZYBA_NEusQ7BN-D0wvZs7ArsRfi0.EP3Php97kjdMCs*/
```

**Reply by j-px** [reply]
AHH I FOUND THE ISSUE. The currently latest GitHub Actions-built build has a bug where it duplicates e.g. `// …` to `/// …` and `/* …` to `/** …` when formatting code. Commit 85297e180abeb244e79fb6ff4384f2de7ea0119c contains a fix for that.

**Reply by ilyassesalama** [reply]
Cool, it works. Now we can call this issue as closed.
